### PR TITLE
Ty/es lint

### PR DESCRIPTION
**Pull request by AlephNotation**
Adding a linter for better code quality

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -53,10 +54,14 @@
...
   },
   "devDependencies": {
+    "@eslint/js": "^9.6.0",
     "@jest/globals": "^29.7.0",
     "@types/debug": "^4.1.12",
     "@types/supertest": "^6.0.2",
     "@types/yargs": "^17.0.32",
+    "eslint": "^9.6.0",
+    "globals": "^15.8.0",
+    "husky": "^9.0.11",
```
Comment (matildepark) [inline package.json:64]: we added husky but don't have any pre-commit hooks?

Comment (AlephNotation) [reply package.json:65]: 😅

Comment (AlephNotation) [reply package.json:65]: fixed now

File: tests/browser/page/browse.test.ts [hunk 2]
```diff
@@ -42,7 +42,7 @@ describe("Page", () => {
       }
     );
 
-    // @ts-ignore
+    // @ts-expect-error - Throwing some weird type inference error due to ObjectiveFailed
```
Comment (matildepark) [inline tests/browser/page/browse.test.ts:45]: fucking unions

**Review by matildepark — APPROVED** [review]

File: tests/browser/page/browse.test.ts (hunk 2 above)
Comment (AlephNotation) [reply tests/browser/page/browse.test.ts:45]: I need to figure out a good way around this

Comment (matildepark) [reply tests/browser/page/browse.test.ts:45]: where the shape changes depending on what value it is? you do it by specifying it as T and conditionally defining it. See https://github.com/hdresearch/ts-sdk/blob/ce2daccd9c490acf8fbb54ef0b961efc31ce5335/src/types/index.ts#L36 and https://github.com/hdresearch/ts-sdk/blob/ce2daccd9c490acf8fbb54ef0b961efc31ce5335/src/types/index.ts#L5

**Pull request merged**